### ISSUE TITLE
“Connected Apps” option not displayed in “Settings” page when user restore account by Resetting Valora earlier

### ISSUE THREAD

**Issue by ValoraQA** [valora-inc/wallet#1366]
**Frequency:** 100%
**Repro on build version:** Android internal build V1.21.0, Playstore build V1.20.4
**Does Not Repro on build version:** Test Flight Build V1.21.0
**Repro on devices:** Redmi Note 8 (10.0), One Plus 7T (11.0), Redmi Note 8 (9.0), Samsung Galaxy F41  (11.0)
**Does not Repro on devices:** iPhone12 (15), iPhone 6+ (12.4.5), iPhone SE (14.2), iPhone SE (13.6.1)
**Testing Account:** (515) 808 5659/[Recovery Phrase](https://docs.google.com/spreadsheets/d/1gXY1I4PBkT9f1uB3zxWgUUJJ1zjVeJcqcWLjiIls1f0/edit#gid=0&range=B71)
**Pre-condition:** 
1. Make sure that the user should have a Restored account by skipping the phone number. 
2. Also make sure that user should have connected any wallet by scanning QR code from desktop & should able to see count in “Connected Apps” section from Settings
**Repro Steps:**
1. Launch the application & redirect to the “Settings” page
2. Tap on the “Reset Valora” option & reset the Valora account
3. Now again restore the wallet with the same account  by skipping the phone number
4. Now tap on “Hamburger Menu” & redirect to the “Settings” page
5. Observe Contents of the Settings page
**Current Behavior:** “Connected Apps” option not displayed in “Security and Data” section from Settings
**Expected Behavior:** “Connected Apps” option should be displayed on the “Settings” page
**Investigation:** Also we have observed in iOS that users able to see the “Connected Apps” section with count as “zero” after restoring the account 
**Impact:** User gets blocked to connect with the wallet as an option  not displaying in Settings
**Attachment:** [“ConnectedApps”OptionNotDisplayed](https://drive.google.com/file/d/1n0QNbKoE2F9jw9Pt1c5rUOI0QvSt4QhD/view?usp=sharing)

**Comment by MuckT** [comment]
@ValoraQA this looks to have been a configuration issue with the remote config, Please retest and close if no longer reproducing.

**Comment by ValoraQA** [comment]
Hey, @MuckT We have retested it on Android internal build V1.21.0 & observed that issue is still reproduced.
**Verified On Devices:** Redmi Note 8 (9.0)
**Attachment:** [AfterRestoringAccountWhichResetEarlier](https://drive.google.com/file/d/121eP1gfp5XPtU8o8TdgCLPio7FKCoUr9/view?usp=sharing)

**Comment by MuckT** [comment]
@ValoraQA I have retested on mainnet and I am no longer seeing this issue. Directly after restore I am able to view the connect app tab is present; Please retest on mainnet.

**After restore**
![Simulator Screen Shot - iPhone 11 - 2021-11-08 at 09 35 43](https://user-images.githubusercontent.com/26950305/140790838-1693cf30-626f-4342-8163-b4c771295380.png)

**Comment by MuckT** [comment]
@jeanregisser @gnardini could we set the `walletConnectV1Enabled` as `true` in `packages/mobile/src/firebase/featureFlagDefaults.ts` to address this issue?

**Comment by MuckT** [comment]
@jeanregisser @gnardini this looks like a larger issue with fetching the remote config from reset accounts. In testing I am not able to successfully complete phone verification on an account that was reset and restored.

```
'KomenciKit/deployWallet attempt#1 error: ', [ [InvalidDeployer: Deployer is not allowed] ]
```

**Comment by MuckT** [comment]
## Workaround
1. Uninstall the App.
2. Reinstall and restore.

**Comment by MuckT** [comment]
Originally appeared in 1.18.x.

**Comment by ValoraQA** [comment]
Hey, @MuckT  We are able to repro this issue on the previous play store build V1.20.4.
**Verified on Devices:** Redmi Note 8 (9.0)
FYI: Currently we are not able to check it on V1.18.x build as we don't have this build right now

**Comment by gnardini** [comment]
I think this change: https://github.com/valora-inc/wallet/pull/1388/files#diff-f8ce68716c543a1d8aca4b1aad9bc1d28e4dccf5f219688cb463918c5b860bbaL246 might help with this issue. 

Can we try again with the current latest version? I think a new build will be put out shortly.

**Comment by gnardini** [comment]
> @jeanregisser @gnardini could we set the walletConnectV1Enabled as true in packages/mobile/src/firebase/featureFlagDefaults.ts to address this issue?

Yes, I think that's safe to do

**Comment by jeanregisser** [comment]
The general problem is remote config stuff are not fetch after a reset. We should re-fetch it when this happens to avoid all these issues.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified this issue on latest Android Internal Build V1.22.1 & observed that the issue is still reproducible.

**Devices:** Redmi Note 8 (10.0)
CC: @gnardini 
Thanks..!!

**Comment by MuckT** [comment]
This should be fixed once #1503 is merged into main. Please retest on the next release or hot-fix.

**Comment by ValoraQA** [comment]
> This should be fixed once #1503 is merged into main. Please retest on the next release or hot-fix.

Sure @MuckT we will retest it when we got the next release/hotfix & will update you here. Thanks..!!

**Comment by MuckT** [comment]
@ValoraQA please retest on 1.22.4.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified this issue using Android Playstore Build V1.22.4 & observed that the issue is fixed now. "Connected Apps" option is displayed in settings screen when user restore account by resetting account earlier.

**Observation:**
- [ ] Connected dapp option is not displayed in connected apps section after we reset the account and restore account using same recovery phrase.

**Device:** Samsung Galaxy A8+(8.0), Redmi Note 8 (11.0)
**Attachment:** [Connected App option appears.mp4](https://drive.google.com/file/d/11DjoASF2BhTzjo4GtD2uqRHg64eTjXT9/view?usp=sharing)  
Thanks..!!

**Comment by MuckT** [comment]
> We have verified this issue using Android Playstore Build V1.22.4 & observed that the issue is fixed now. "Connected Apps" option is displayed in settings screen when user restore account by resetting account earlier. 

Closing the issue as it is no longer reproducible.
